serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/dto/DTOConverter.java

```diff
@@ -166,7 +168,7 @@ public SystemResponseDTO convertSystemTripletToDTO(final Triple<System, List<Sys
 		Assert.notNull(entity, "entity is null");
 		Assert.notNull(entity.getLeft(), "the System in the triplet is null");
 		Assert.isTrue(!Utilities.isEmpty(entity.getMiddle()), "the system address list in the triplet is null");
-		Assert.isTrue(entity.getRight() == null || entity.getRight().getValue() != null, "the device address list in the triplet is null");
+		Assert.isTrue(entity.getRight() == null || entity.getRight().getValue() != null && !Utilities.isEmpty(entity.getRight().getValue()), "the device address list in the triplet is null or empty");
```

Comment:
change this 
`entity.getRight().getValue() != null && !Utilities.isEmpty(entity.getRight().getValue())`
to this
`!Utilities.isEmpty(entity.getRight().getValue())`
because this is already contains the null check